Source/plrmsg.cpp

```diff
@@ -9,32 +9,45 @@
 
 #include <fmt/format.h>
 
-#include "DiabloUI/ui_flags.hpp"
 #include "control.h"
 #include "engine/render/text_render.hpp"
 #include "inv.h"
 #include "utils/language.h"
-#include "utils/stdcompat/string_view.hpp"
 #include "utils/utf8.hpp"
 
 namespace devilution {
 
 namespace {
 
-#define PMSG_COUNT 8
+struct PlayerMessage {
+	/** Time message was recived */
+	Uint32 time;
+	/** The default text color */
+	UiFlags style;
+	/** The text message to display on screen */
+	std::string text;
+	/** Portion of text that should be rendered in gold */
+	std::string from;

```

Comment:
This is implied to be a span of the full text string based on the comment, and is really required for correct rendering given the way the gold section is overdrawn the full line. I think this should be a `string_view` pointing to the first `n` characters of `text`. Could be done in a constructor or in the `SendPlrMsg()` function.